pkg/util/strategicpatch/patch.go

```diff
@@ -240,15 +602,20 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	return original, nil
 }
 
-// This panics if any element of the slice is not a map.
+// This method no longer panics if any element of the slice is not a map.
```

Comment:
Any reason to change this? I think I made it panic deliberately since it should be indicating programmer error, not user error, since it should not be possible based on its surrounding code to call it with a slice that has a different type than `value`.